[MPS] Enable `conj` and `conj_physical`

malfet opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #119670
* __->__ #119669
* #119681

Former is only on MacOS 14+, but at least on older MacOSes it would raise an exception rather than returning non-conjugated tensor

Preliminary step for enabling FFT ops (without it `ifft` would never work)

malfet committed:
Enable `conj_physical`

Preliminary step for enabling FFT ops (without it `ifft` would never work)

[ghstack-poisoned]

malfet committed:
Update on "Enable `conj_physical`"


Preliminary step for enabling FFT ops (without it `ifft` would never work)

[ghstack-poisoned]

malfet committed:
Update on "Enable `conj_physical`"


Preliminary step for enabling FFT ops (without it `ifft` would never work)

[ghstack-poisoned]

malfet committed:
Update on "Enable `conj` and `conj_physical`"


Former is only on MacOS 14+, but at least on older MacOSes it would now raise an exception rather than pretend that it works.

Preliminary step for enabling FFT ops (without it `ifft` would never work)

[ghstack-poisoned]

malfet committed:
Update on "Enable `conj` and `conj_physical`"


Former is only on MacOS 14+, but at least on older MacOSes it would now raise an exception rather than pretend that it works.

Preliminary step for enabling FFT ops (without it `ifft` would never work)

[ghstack-poisoned]

malfet committed:
Update on "Enable `conj` and `conj_physical`"


Former is only on MacOS 14+, but at least on older MacOSes it would now raise an exception rather than pretend that it works.

Preliminary step for enabling FFT ops (without it `ifft` would never work)

[ghstack-poisoned]

albanD commented on aten/src/ATen/native/mps/operations/UnaryOps.mm:
```diff
@@ -491,4 +495,17 @@ static void cumulative_op_impl(const Tensor& self,
+  TORCH_CHECK(self.is_complex());
+  if (!mps::supportsComplex()) {
+    result = self;
+    at::imag(result) = at::neg(at::imag(self));
```
These assignments need to be copy_

malfet committed:
Update on "[MPS] Enable `conj` and `conj_physical`"


Former is only on MacOS 14+, but at least on older MacOSes it would raise an exception rather than returning non-conjugated tensor

Preliminary step for enabling FFT ops (without it `ifft` would never work)

[ghstack-poisoned]

albanD reviewed (approved):
I guess they don't have to, my bad :D

malfet committed:
Update on "[MPS] Enable `conj` and `conj_physical`"


Former is only on MacOS 14+, but at least on older MacOSes it would raise an exception rather than returning non-conjugated tensor

Preliminary step for enabling FFT ops (without it `ifft` would never work)

[ghstack-poisoned]

malfet committed:
Update on "[MPS] Enable `conj` and `conj_physical`"


Former is only on MacOS 14+, but at least on older MacOSes it would raise an exception rather than returning non-conjugated tensor

Preliminary step for enabling FFT ops (without it `ifft` would never work)

[ghstack-poisoned]

malfet committed:
Update on "[MPS] Enable `conj` and `conj_physical`"


Former is only on MacOS 14+, but at least on older MacOSes it would raise an exception rather than returning non-conjugated tensor

Preliminary step for enabling FFT ops (without it `ifft` would never work)

[ghstack-poisoned]

malfet commented:
@pytorchbot merge -f "Lint and MPS tests are green"
